Update google-logins.md by using Google.Apis.Auth.AspNetCore3 package (#33710)
dotnet/AspNetCore.Docs: aspnetcore/security/authentication/social/google-logins.md

---
# Google external login setup in ASP.NET Core

By [Valeriy Novytskyy](https://github.com/01binary), [Rick Anderson](https://twitter.com/RickAndMSFT) and [Sharaf Abacery](https://github.com/sharafabacery)

This tutorial shows you how to enable users to sign in with their Google account using the ASP.NET Core project created on the [previous page](xref:security/authentication/social/index).

## Create the Google OAuth 2.0 Client ID and secret

> * Follow the guidance in [Integrating Google Sign-In into your web app](https://developers.google.com/identity/gsi/web/guides/overview) (Google documentation)

Reviewer: that goes to:
 Google Identity Services is migrating to FedCM APIs. Follow the [migration guide](https://developers.google.com/identity/gsi/web/guides/fedcm-migration) to review potential changes and avoid negative impacts for user sign-in to your website.

Shouldn't we point to *** FedCM APIs*** ???

Author: ok,I see your point ,be more  general for need to sign-in

---
# Google external login setup in ASP.NET Core

> By [Valeriy Novytskyy](https://github.com/01binary) , [Rick Anderson](https://twitter.com/RickAndMSFT) and [Sharaf Abacery](https://github.com/sharafabacery)

Reviewer: ```suggestion
By [Valeriy Novytskyy](https://github.com/01binary), [Rick Anderson](https://twitter.com/RickAndMSFT) and [Sharaf Abacery](https://github.com/sharafabacery)
```

This tutorial shows you how to enable users to sign in with their Google account using the ASP.NET Core project created on the [previous page](xref:security/authentication/social/index).

## Create the Google OAuth 2.0 Client ID and secret

> * Follow the guidance in [Integrating Google Sign-In into your web app (New)](https://developers.google.com/identity/gsi/web/guides/overview) (Google documentation)
> * [Integrating Google Sign-In into your web app (Deprecated)](https://developers.google.com/identity/sign-in/web/sign-in) (Google documentation). (Deprecated)

Reviewer: ```suggestion
* Follow the guidance in [Integrating Google Sign-In into your web app](https://developers.google.com/identity/gsi/web/guides/overview) (Google documentation)
```

You can manage your API credentials and usage in the [API Console](https://console.developers.google.com/apis/dashboard).

> ## Configure Google authentication (First Edition)

Reviewer: If `(First Edition)` is deprecated, delete it. Just list the latest instructions and delete everything deprecated.

Author: I edited

Author: thank you for help

## Configure Google authentication

* Add the [`Google.Apis.Auth.AspNetCore3`](https://www.nuget.org/packages/Google.Apis.Auth.AspNetCore3) NuGet package to the app.
* Add the Authentication service to the `program.cs`:
* Follow [`Add Authtication for asp.net app`](https://developers.google.com/api-client-library/dotnet/guide/aaa_oauth#configure-your-application-to-use-google.apis.auth.aspnetcore3)

[!INCLUDE [default settings configuration](includes/default-settings2-2.md)]

## Sign in with Google
* Get a link to the libary at [google developer library link ](https://developers.google.com/identity/gsi/web/guides/client-library) to get link of library.
* Then go to [google developer button genration ](https://developers.google.com/identity/gsi/web/tools/configurator)
* Setup your Controller to match with ` data-login_uri="{HostName}/{ControllerName}/{actionName}" ` attrbute because after success login it will forward you to that link.

> * Create controller and action takes one argement  `string credential` because that what google return when complete login process.

Reviewer 2: ```suggestion
* Create a controller and action takes one argument `string credential`, because that is what Google returns when the login process is complete.
```

> * Verify `credential` by using this line of code

Reviewer 2: ```suggestion
* Verify the `credential` using the following line of code:
```

`GoogleJsonWebSignature.Payload payload = await GoogleJsonWebSignature.ValidateAsync(credential);`

> * Here you  get all information about login user you can store it in database.

Reviewer 2: ```suggestion
* This will retrieve the available information about the logged-in user, which could then be stored in a database.
```

> [View or download the completed project](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/security/authorization/social/GoogleAuthExample) ([how to download](xref:index#how-to-download-a-sample)).

Reviewer 2: ```suggestion
``

Per Rick's feedback I'll remove this link to the example you provided.  Thanks for going the extra mile however to try to provide one.

* Add the Authentication service to the `program.cs`:
* Follow [`Add Authtication for asp.net app`](https://developers.google.com/api-client-library/dotnet/guide/aaa_oauth#configure-your-application-to-use-google.apis.auth.aspnetcore3)

[!INCLUDE [default settings configuration](includes/default-settings2-2.md)]

## Sign in with Google
* Get a link to the libary at [google developer library link ](https://developers.google.com/identity/gsi/web/guides/client-library) to get link of library.
* Then go to [google developer button genration ](https://developers.google.com/identity/gsi/web/tools/configurator)
* Setup your Controller to match with ` data-login_uri="{HostName}/{ControllerName}/{actionName}" ` attrbute because after success login it will forward you to that link.
* Create controller and action takes one argement  `string credential` because that what google return when complete login process.
* Verify `credential` by using this line of code
`GoogleJsonWebSignature.Payload payload = await GoogleJsonWebSignature.ValidateAsync(credential);`
* Here you  get all information about login user you can store it in database.

> [Complete Simple working app](https://github.com/sharafabacery/GoogleAuthticationExample)

Reviewer: See https://github.com/dotnet/AspNetCore.Docs.Samples/issues/264

Author: I add the sample
